src/value.cc

```diff
@@ -625,14 +635,30 @@ gpointer V8ToGHash (GITypeInfo *type_info, Local<Value> value) {
     return hash_table;
 }
 
-bool V8ToGIArgument(GIBaseInfo *gi_info, GIArgument *arg, Local<Value> value) {
+bool V8ToGIArgument(GIBaseInfo *gi_info, GIArgument *arg, Local<Value> value, GITransfer transfer) {
     GIInfoType type = g_base_info_get_type (gi_info);
 
     switch (type) {
     case GI_INFO_TYPE_BOXED:
     case GI_INFO_TYPE_STRUCT:
     case GI_INFO_TYPE_UNION:
         arg->v_pointer = PointerFromWrapper(value);
+        if (transfer == GI_TRANSFER_EVERYTHING) {
+            GType gtype = g_registered_type_info_get_g_type (gi_info);
+            size_t size;
+
+            if (gtype != G_TYPE_NONE) {
+                arg->v_pointer = g_boxed_copy (gtype, arg->v_pointer);
+            } else if ((size = Boxed::GetSize(gi_info)) != 0) {
+                void *boxedCopy = malloc(size);
+                memcpy(boxedCopy, arg->v_pointer, size);
+                arg->v_pointer = boxedCopy;
+            } else {
+                // XXX what else can we do? Should we just abort?
+                WARN("Cannot copy a boxed object of type '%s'; memory corruption might occur.",
+                        g_base_info_get_name(gi_info));
+            }
+        }
```

Comment:
This seems to be duplicated from `boxed.cc`, is there a way to avoid duplication by extracting a function?

Reply:
I've de-duplicated it in the next commit. How do you think?

Reply:
Good!